simulator/src/utils/common_instructions.rs

```diff
@@ -171,9 +172,237 @@ macro_rules! matches_bucket {
+    type_kind: &ScryptoTypeKind<LocalTypeId>,
+    type_validation: &TypeValidation<ScryptoCustomTypeValidation>,
+) -> Result<ManifestValueKind, BuildCallArgumentError> {
+    match type_kind {
+        ScryptoTypeKind::Bool => Ok(ManifestValueKind::Bool),
+        ScryptoTypeKind::I8 => Ok(ManifestValueKind::I8),
+        ScryptoTypeKind::I16 => Ok(ManifestValueKind::I16),
+        ScryptoTypeKind::I32 => Ok(ManifestValueKind::I32),
+        ScryptoTypeKind::I64 => Ok(ManifestValueKind::I64),
+        ScryptoTypeKind::I128 => Ok(ManifestValueKind::I128),
+        ScryptoTypeKind::U8 => Ok(ManifestValueKind::U8),
+        ScryptoTypeKind::U16 => Ok(ManifestValueKind::U16),
+        ScryptoTypeKind::U32 => Ok(ManifestValueKind::U32),
+        ScryptoTypeKind::U64 => Ok(ManifestValueKind::U64),
+        ScryptoTypeKind::U128 => Ok(ManifestValueKind::U128),
+        ScryptoTypeKind::String => Ok(ManifestValueKind::String),
+        ScryptoTypeKind::Array { .. } => Ok(ManifestValueKind::Array),
+        ScryptoTypeKind::Tuple { .. } => Ok(ManifestValueKind::Tuple),
+        ScryptoTypeKind::Enum { .. } => Ok(ManifestValueKind::Enum),
+        ScryptoTypeKind::Map { .. } => Ok(ManifestValueKind::Map),
+        ScryptoTypeKind::Custom(ScryptoCustomTypeKind::Decimal) => {
+            Ok(ManifestValueKind::Custom(ManifestCustomValueKind::Decimal))
+        }
+        ScryptoTypeKind::Custom(ScryptoCustomTypeKind::PreciseDecimal) => Ok(
+            ManifestValueKind::Custom(ManifestCustomValueKind::PreciseDecimal),
+        ),
+        ScryptoTypeKind::Custom(ScryptoCustomTypeKind::NonFungibleLocalId) => Ok(
+            ManifestValueKind::Custom(ManifestCustomValueKind::NonFungibleLocalId),
+        ),
+        ScryptoTypeKind::Custom(ScryptoCustomTypeKind::Reference) => {
+            Ok(ManifestValueKind::Custom(ManifestCustomValueKind::Address))
+        }
+        ScryptoTypeKind::Custom(ScryptoCustomTypeKind::Own)
+            if matches_bucket!(type_validation)
+                || matches_bucket!(type_validation, FUNGIBLE_BUCKET_BLUEPRINT)
+                || matches_bucket!(type_validation, NON_FUNGIBLE_BUCKET_BLUEPRINT) =>
+        {
+            Ok(ManifestValueKind::Custom(ManifestCustomValueKind::Bucket))
+        }
+
+        ScryptoTypeKind::Custom(ScryptoCustomTypeKind::Own)
+            if matches!(
+                type_validation,
+                TypeValidation::Custom(ScryptoCustomTypeValidation::Own(OwnValidation::IsProof))
+            ) =>
+        {
+            Ok(ManifestValueKind::Custom(ManifestCustomValueKind::Proof))
+        }
+        _ => Err(BuildCallArgumentError::UnsupportedType(type_kind.clone())),
+    }
+}
+
+// Splits argument tuple or array elements delimited with "," into vector of elements.
+// Elements with brackets (round or square) are taken as is (even if they may include ",")
+fn split_argument_tuple_or_array(argument: &str) -> Result<Vec<String>, BuildCallArgumentError> {
```

Comment:
... Although we could also consider adding string support with `"` to the below parser (and the string parser would discard start and stopping `"`) -- i.e. we could have an "is in string" flag, and avoid interpreting any other characters until we see a `"`.

Reply:
Yeah, surrounding strings with `"` would make things easier, but it would be quite different of other places in `resim`, where string could be supplied. Not sure if want it...

Reply:
I guess I was thinking that it could be optional, so `abc`, `"abc"` and `"abc [ asda"` are all acceptable for a string. But it does further complicate the splitter/parser!